mongokit/apply_test.go

```diff
@@ -654,3 +654,63 @@ func TestApplyCurrentDate(t *testing.T) {
 		Removed: map[string]interface{}{},
 	}, changes)
 }
+
+func TestApplyPush(t *testing.T) {
```

Comment:
Could you combine the three applyTests into one? You can call `fn` multiple times and  start by adding an element to the empty doc and then update that and so on. This way we do not allocate unnecessary collections.

Reply:
I don't understand how can i start by adding an element to the empy doc.
Do you mean like this code?
```go
		// add element
		fn(bson.M{
			"$push": bson.M{
				"foo": "bag",
			},
		}, bsonkit.Convert(bson.M{
			"foo": bson.A{"bag"},
		}))

		// add element
		fn(bson.M{
			"$push": bson.M{
				"foo": "foo",
			},
		}, bsonkit.Convert(bson.M{
			"foo": bson.A{"bag", "foo"},
		}))

		// remove foo
		fn(bson.M{
			"$set": bson.M{
				"foo": "foobar",
			},
		}, bsonkit.Convert(bson.M{
			"foo": "foobar",
		}))

		// add element
		fn(bson.M{
			"$push": bson.M{
				"foo": "foobaz",
			},
		}, bsonkit.Convert(bson.M{
			"foo": bson.A{"foobar", "foobaz"},
		}))
```

Reply:
Yes. Theoretically, you could also have multiple fields in the bootstrapped document and perform multiple tests at once:

```go
applyTest(t, false, bson.M{
 		"foo": bson.A{
 			"bar",
 			"baz",
 		},
               "string": "foo", // <---
 	}, func(fn func(bson.M, interface{})) {
 		// TODO: Append test that tries to $push to the string field
 	})
```